apps/explorer/src/ui/Search.tsx

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { Combobox } from '@headlessui/react';
+import { Search16 } from '@mysten/icons';
+
+import { LoadingSpinner } from './LoadingSpinner';
+import { Text } from './Text';
+
+type SearchResult = {
+    id: number;
+    label: string;
+};
+
+export interface SearchProps {
+    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    onSelectResult: (result: SearchResult) => void;
+    placeholder?: string;
+    isLoading: boolean;
+    name?: string;
+    options?: Record<string, SearchResult[]>;
+    value?: SearchResult;
```

Comment:
`value` and `inputValue` I think can be slightly confusing, maybe worth explicitly calling this something like `queryValue`?

Reply:
@mamos-mysten Lol to be clear I meant having both of these, and removing `value` because it's unclear what it is.